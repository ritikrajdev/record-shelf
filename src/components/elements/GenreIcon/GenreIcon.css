.genre-icon {
  position: relative;
  display: flex;
  align-items: center;
}

.genre-icon img {
  width: 150px;
  height: 150px;
}

.genre-icon > div {
  transform: translateX(-5rem);
}

@media screen and (max-width: 350px) {
  .genre-icon img {
    width: 100px;
    height: 100px;
  }

  .genre-icon > div {
    transform: translateX(-3rem);
  }

  .genre-icon button {
    width: 200px;
    text-overflow: ellipsis;
    overflow: hidden;
    -webkit-line-clamp: 1;
  }
}
