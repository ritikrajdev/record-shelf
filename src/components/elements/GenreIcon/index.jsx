import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import RoundedButton from '../RoundedButton';

import defaultGenereIcon from '../../../assets/images/genre-pop.png';

import './GenreIcon.css';

export default function GenreIcon({ genre }) {
  const [genreIcon, setGenreIcon] = useState(defaultGenereIcon);

  useEffect(() => {
    import(`../../../assets/images/genre-${genre}.png`)
      .then((module) => setGenreIcon(module.default))
      .catch(console.error);
  });

  return (
    <div className='genre-icon'>
      <img src={genreIcon} alt={genre} />
      <div>
        <RoundedButton>{genre}</RoundedButton>
      </div>
    </div>
  );
}

GenreIcon.propTypes = {
  genre: PropTypes.string,
};
