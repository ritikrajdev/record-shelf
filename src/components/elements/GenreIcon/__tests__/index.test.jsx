import { render } from '@testing-library/react';
import React from 'react';
import GenreIcon from '..';

describe('GenereIcon', () => {
  it('should render correctly', () => {
    const screen = render(<GenreIcon genre={'pop'} />);
    expect(screen.asFragment()).toMatchSnapshot();
  });
});
