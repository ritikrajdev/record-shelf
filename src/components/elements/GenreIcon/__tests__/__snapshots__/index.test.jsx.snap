// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenereIcon should render correctly 1`] = `
<DocumentFragment>
  <div
    class="genre-icon"
  >
    <img
      alt="pop"
      src="genre-pop.png"
    />
    <div>
      <button
        class="rounded-button"
      >
        pop
      </button>
    </div>
  </div>
</DocumentFragment>
`;
